src/main/java/com/saasquatch/jsonschemainferrer/JsonSchemaInferrer.java

```diff
@@ -153,7 +136,8 @@ private ObjectNode processObjects(@Nonnull Collection<ObjectNode> objectNodes) {
           .map(this::preProcessSample).collect(Collectors.toList());
       final ObjectNode newProperty = newObject();
       handleDescriptionGeneration(newProperty, fieldName);
-      final Set<ObjectNode> anyOfs = getAnyOfsFromSamples(processedSamples);
+      final String objectPath = path + "." + fieldName;
```

Comment:
What if the field name has special characters?

Reply:
JsonPath uses "foo.bar" but also allows `foo["bar"]`. I can change it to that to support special characters. Alternatively, instead of a `String`, it can be a `String[]`/`List<String>` if that would be preferred.

Reply:
This pull request has prompted me to look into the [JSON schema library](https://github.com/networknt/json-schema-validator) I'm using, which I know also generates JSON paths, and I actually discovered a related bug, and I reported it [here](https://github.com/networknt/json-schema-validator/issues/750).

For the record, [this](https://github.com/networknt/json-schema-validator/blob/8107dfea7b2c225fdf5fd895f53939bc5c4d620d/src/main/java/com/networknt/schema/PathType.java#L17) is their code for generating JSON paths.

I'm not a JSON path expert, and all the examples I can find online use single quotes (i.e. something like `$['foo']`) instead of double quotes, and I haven't found any explicit confirmation suggesting that the part between the square brackets can be any valid JSON string. If it's the case, then this is an easy problem to solve because we can just use Jackson like this

```java
"[" + JsonNodeFactory.instance.textNode(token) + "]"
```

Reply:
I implemented it as you stated, also created an additional test to see if the path is resulted properly.